src/util/utf8.ts

```diff
@@ -17,7 +17,7 @@
 
 const decoder = new TextDecoder('utf-8');
 /** @ignore */
-export const decodeUtf8 = (buffer?: BufferSource) => decoder.decode(buffer);
+export const decodeUtf8 = (buffer?: Parameters<InstanceType<typeof TextDecoder>['decode']>[0]) => decoder.decode(buffer);
```

Comment:
I think this is also an option?
```suggestion
export const decodeUtf8 = decoder.decode.bind(decoder);
```